File: frontend/lib/models/payroll.dart
class Payroll {
  final String id;
  final String employeeId;
  final int month;
  final int year;
  final int workingDays;
  final int presentDays;
  final int absentDays;
  final double perDaySalary;
  final double payableSalary;
  final String status;

  Payroll({
    required this.id,
    required this.employeeId,
    required this.month,
    required this.year,
    required this.workingDays,
    required this.presentDays,
    required this.absentDays,
    required this.perDaySalary,
    required this.payableSalary,
    required this.status,
  });

  factory Payroll.fromJson(Map<String, dynamic> json) {
    return Payroll(
      id: json['_id'] ?? json['id'] ?? '',
      employeeId: json['employeeId'] ?? '',
      month: json['month'] ?? 1,
      year: json['year'] ?? DateTime.now().year,
      workingDays: json['workingDays'] ?? 0,
      presentDays: json['presentDays'] ?? 0,
      absentDays: json['absentDays'] ?? 0,
      perDaySalary: (json['perDaySalary'] ?? 0).toDouble(),
      payableSalary: (json['payableSalary'] ?? 0).toDouble(),
      status: json['status'] ?? 'GENERATED',
    );
  }
}

File: backend/src/models/Payroll.js
import mongoose from "mongoose";

const payrollSchema = new mongoose.Schema(
  {
    employeeId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "Employee",
      required: true,
    },
    month: Number,
    year: Number,

    workingDays: Number,
    presentDays: Number,
    absentDays: Number,

    perDaySalary: Number,
    payableSalary: Number,

    status: {
      type: String,
      default: "GENERATED", // GENERATED | PAID
    },
  },
  { timestamps: true },
);

export default mongoose.model("Payroll", payrollSchema);
